src/transpiler_hook.ts

```diff
@@ -0,0 +1,29 @@
+class TranspilerHook {
```

Comment:
`TranspilerHook` is probably not the best name. If you want to name it like this then it would be `TranspilerHooks`

Reply:
And using an interface here would probably be better. The only downside I see, is that we would need an empty implementation.

Reply:
If we use an interface, every hook class needs to implement all hook methods. I don't think this is a good idea.

Reply:
I don't think that this would be a big problem because in this case we want to describe the contract that the hook methods have to fulfil.